lib/dev/frame-script.js

```diff
@@ -110,6 +113,8 @@ const observer = {
 
 observerService.addObserver(observer, "content-document-interactive", false);
 observerService.addObserver(observer, "content-document-loaded", false);
-observerService.addObserver(observer, "content-page-hidden", false);
+observerService.addObserver(observer, "chrome-document-interactive", false);
+observerService.addObserver(observer, "chrome-document-loaded", false);
+addEventListener("unload", observer, false);
```

Comment:
Shouldn't you remove this listener at some point? Same question for observerService?
Or is this frame script global and want to dispatch events for all document living in the tab?